Spot/Spot.swift

```diff
@@ -66,9 +68,15 @@ extension UIWindow {
         if let initialViewController = storyboard.instantiateInitialViewController() as? OrientationLockNavigationController {
             initialViewController.orientationToLock = UIDevice.current.orientation
             let window = UIApplication.shared.keyWindow
-            window?.rootViewController?.present(initialViewController, animated: true, completion: nil)
-            if let screenshotViewController = initialViewController.topViewController as? SpotViewController {
-                screenshotViewController.screenshot = screenshot
+            if var topController = window?.rootViewController {
+                while let presentedViewController = topController.presentedViewController {
+                    topController = presentedViewController
+                }
+                // topController should now be your topmost view controller
+                topController.present(initialViewController, animated: true, completion: nil)
```

Comment:
That means that you might have already done a screenshot plus much unnecessary loading of files that you don't actually need. By keeping state, you avoid going through all those calls